src/mem/epoch/garbage.rs

```diff
@@ -23,10 +20,12 @@ impl Bag {
         if size > 0 {
             self.0.push(Item {
                 ptr: elem as *mut u8,
-                size: size,
-                align: mem::align_of::<T>(),
+                free: free::<T>,
             })
         }
+        unsafe fn free<T>(t: *mut u8) {
+            drop(Vec::from_raw_parts(t as *mut T, 0, 1));
```

Comment:
This violates the `Vec` docs, doesn't it? (where `from_raw_parts` should only be passed the result of `into_raw_parts`) I think at least some kind of `// HACK` comment is in order, if you really want to get rid of `heap::deallocate`. Also the extra level of indirection introduced by calling `free` for each element may turn out to have perf impact, have you measured any decrease in the benchmarks? My 2c.